### ISSUE TITLE
Request for training resource details: GPU VRAM usage and training duration

### ISSUE THREAD

**Issue by ChenyiXu04** [jayLEE0301/TraceGen#10]
**Thank you for your excellent work!** 🙏

First of all, congratulations on the amazing TraceGen paper! The idea of modeling in compact 3D trace-space is truly brilliant — it achieves strong cross-embodiment transfer with only 0.67B parameters and 50–600× faster inference than video-based world models. The TraceForge pipeline and the 123K dataset are also extremely valuable contributions to the community.

I'm currently reproducing the model using the official code[](https://github.com/tracegen/...) and have successfully initialized TraceGen (0.83B total params, 232M trainable, CogVideoX flow decoder + frozen DINOv3/SigLIP/T5). The model loads correctly on CUDA with `torch.compile`.

However, the paper and codebase do not provide any details about **training resource consumption**. Could you kindly share the following information? This would be extremely helpful for the community to better understand the practicality of pretraining:

### Questions:
1. **GPU Memory (VRAM) usage** during full pretraining on the TraceForge-123K dataset (1.8M triplets)?
   - Peak VRAM per GPU?
   - What batch size / gradient accumulation / mixed precision (fp16/bf16) was used?

2. **Training duration** (wall-clock time)?
   - Total GPU-hours (or days on your cluster)?
   - Number of epochs / total steps?
   - Hardware used (e.g. 8×A100 80GB, H100, etc.)?

3. Any recommended config for single-GPU or limited-resource training (e.g. reduced batch size, gradient checkpointing, or training on a subset)?

These details are not mentioned in the paper (Appendix C only discusses frozen encoders) or the current README, and would greatly help researchers and practitioners who want to pretrain or fine-tune TraceGen.

Thank you again for releasing such a clean and well-structured codebase — the logging and modular design are really appreciated!

Looking forward to your reply.

Best regards,  
Chenyi Xu

**Comment by adreamwu** [comment]
Hi @ChenyiXu04 , I noticed you closed this issue.

Did you figure out the training duration requirements? I'm curious about the resources needed for TraceGen as well. Thanks!

**Comment by ChenyiXu04** [comment]
> 你好[@ChenyiXu04](https://github.com/ChenyiXu04)我注意到您已关闭此问题。
> 
> 你弄清楚培训时长要求了吗？我也很好奇 TraceGen 需要哪些资源。谢谢！

Hi, according to the official configuration provided in the repository:
```
export CUDA_VISIBLE_DEVICES=0
python train.py \
  --config cfg/train.yaml \
  --override \
  train.batch_size=6 \
  train.lr_decoder=1.5e-4 \
  model.decoder.num_layers=12 \
  model.decoder.num_attention_heads=16 \
  model.decoder.latent_dim=768 \
  data.num_workers=4 \
  hardware.mixed_precision=true \
  logging.use_wandb=true \
  logging.log_every=2000
```
I trained the model on a single NVIDIA RTX 4090(24GB). However, the VRAM was not sufficient for ` train.batch_size=6 `, so I reduced it to 4.

With this setup, each epoch takes about 30 minutes. Based on this, I estimate that training the model on a single 4090 would take around 350 hours in total.

This is only a rough estimate and may vary depending on hardware and other settings.
